packages/@aws-cdk/aws-eks/lib/helm-chart/index.py

```diff
@@ -0,0 +1,135 @@
+import subprocess
+import os
+import json
+import logging
+import boto3
+from uuid import uuid4
+from botocore.vendored import requests
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+# these are coming from the kubectl layer
+os.environ['PATH'] = '/opt/helm:/opt/awscli:' + os.environ['PATH']
+
+outdir = os.environ.get('TEST_OUTDIR', '/tmp')
+kubeconfig = os.path.join(outdir, 'kubeconfig')
+
+CFN_SUCCESS = "SUCCESS"
+CFN_FAILED = "FAILED"
+
+def handler(event, context):
+
+    def cfn_error(message=None):
+        logger.error("| cfn_error: %s" % message)
+        cfn_send(event, context, CFN_FAILED, reason=message)
+
+    try:
+        logger.info(json.dumps(event))
+
+        request_type = event['RequestType']
+        props = event['ResourceProperties']
+        physical_id = event.get('PhysicalResourceId', None)
+        name = props['Name']
+        chart = props['Chart']
+        version = props.get('Version', None)
+        namespace = props.get('Namespace', None)
+        repository = props.get('Repository', None)
+        values_text = props.get('Values', None)
+
+        cluster_name = os.environ.get('CLUSTER_NAME', None)
+        if cluster_name is None:
+            cfn_error("CLUSTER_NAME is missing in environment")
+            return
+        
+        subprocess.check_call([ 'aws', 'eks', 'update-kubeconfig',
+            '--name', cluster_name,
+            '--kubeconfig', kubeconfig
+        ])
+
+        # Write out the values to a file and include them with the install and upgrade
+        values_file = None
+        if not request_type == "Delete" and not values_text is None:
+            values = json.loads(values_text)
+            values_file = os.path.join(outdir, 'values.yaml')
+            with open(values_file, "w") as f:
+                f.write(json.dumps(values, indent=2))
+
+        if request_type == 'Create':
+            helm('install', name, chart, repository, values_file, namespace, version)
+        elif request_type == 'Update':
+            helm('upgrade', name, chart, repository, values_file, namespace, version)
```

Comment:
Quick question. It would be nice if the helm chart resource works with EKS clusters that already have helm charts deployed.

If you change these commands to use `helm upgrade --install` instead, then this resource can be used seamlessly with existing helm chart deployments. This will really help us refactor our code 🙏 We have some EKS clusters running with helm charts that we really can't delete.

Reply:
Ok, I've changed this so it will no fail when you try to install a chart which already exists in a cluster. It will simply upgrade the chart with a new revision.